src/VisualStudio/Core/Def/Packaging/PackageInstallerServiceFactory.cs

```diff
@@ -523,44 +456,43 @@ private void ProcessProjectChange(Solution solution, ProjectId projectId)
             }
 
             // Project was changed in some way.  Let's go find the set of installed packages for it.
-            var dteProject = _workspace.TryGetDTEProject(projectId);
-            if (dteProject == null)
+            var projectGuid = _workspace.GetProjectGuid(projectId);
+            if (projectGuid == Guid.Empty)
             {
                 // Don't have a DTE project for this project ID.  not something we can query NuGet for.
                 return;
             }
 
-            var installedPackages = new MultiDictionary<string, string>();
-            var isEnabled = false;
+            var serviceContainer = _serviceProvider.GetService<IBrokeredServiceContainer, SVsBrokeredServiceContainer>();
+            var serviceBroker = serviceContainer.GetFullAccessServiceBroker();
```

Comment:
Make sure to check for a null `serviceContainer` before dereferencing, unless you're calling an extension method that guarantees to throw instead of return null. One of our `IAsyncServiceProvider.GetServiceAsync<S, I>` extension methods does just this. You can tell by its nullable-ref annotations that it never returns null.